test/integration/camera_sensor.cc

```diff
@@ -267,6 +267,61 @@ TEST_F(CameraSensor, CheckThrottle)
   // spawn sensors of various sizes to test speed
   std::string modelName = "camera_model";
   std::string cameraName = "camera_sensor";
+  unsigned int width  = 320;
+  unsigned int height = 240;  // 106 fps
+  double updateRate = 10;
+  ignition::math::Pose3d setPose, testPose(ignition::math::Vector3d(-5, 0, 5),
+      ignition::math::Quaterniond(0, IGN_DTOR(15), 0));
+  SpawnCamera(modelName, cameraName, setPose.Pos(),
+      setPose.Rot().Euler(), width, height, updateRate);
+  sensors::SensorPtr sensor = sensors::get_sensor(cameraName);
+  sensors::CameraSensorPtr camSensor =
+    std::dynamic_pointer_cast<sensors::CameraSensor>(sensor);
+  imageCount = 0;
+  img = new unsigned char[width * height*3];
+  event::ConnectionPtr c = camSensor->Camera()->ConnectNewImageFrame(
+        std::bind(&::OnNewCameraFrame, &imageCount, img,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+  common::Timer timer;
+  timer.Start();
+
+  // time how long it takes to get 50 images @ 10Hz
+  int total_images = 50;
+
+  while (imageCount < total_images)
+    common::Time::MSleep(10);
+  common::Time dt = timer.GetElapsed();
+  double rate = static_cast<double>(total_images)/dt.Double();
+  gzdbg << "timer [" << dt.Double() << "] seconds rate [" << rate << "] fps\n";
+  EXPECT_GT(rate, 7.0);
+  EXPECT_LT(rate, 11.0);
+
+  c.reset();
+  delete [] img;
+}
+
+/////////////////////////////////////////////////
+TEST_F(CameraSensor, CheckThrottleStrictRate)
```

Comment:
So I actually ended up increasing to 0.06 (from 0.02) in another branch... 0.06 being arbitrary for the test numbers on my computer... I thought maybe my computer is slow, but the point of this PR was to make the update rate work regardless of the speed of the machine... so it seems that it isn't quite hitting the rate that's asked of it? Though 488 out of 500 isn't bad, I've gotten as low as 473 before, and that's where 0.06 came from. In normal usage, people might never ask for 500 fps, so it might be okay to raise the tolerance. But then the bit about how this whole lockstep idea is so that any computer can hit the update rate isn't quite satisfied.

Reply:
Raised to 0.06 in c134381

Reply:
ed6fc04 tries to get a time based on the images received, which passed the test 10 times with tolerance lowered to the original 0.02.
I didn't find a place where the timestamp is delivered - it seems `rendering::Camera::ConnectNewImageFrame()` returns the image in a `char *`, and doesn't return a stamp. Another option is `rendering::Camera::LastRenderWallTime()`, but that is wall time and not sim time.
So currently, I am getting the first timestamp manually by checking in the waiting loop for `imageCount == 0`, which means an iteration before the first image is received. It's still not ideal, in that it might be making the test easier to pass.
I'm consistently getting this rate:
`[Dbg] [camera_sensor.cc:354] timer [4.999] seconds rate [500.1] fps`
Let me know if you think this change is reasonable, i.e. not too lenient...